projects/block-level-access-lists-in-reth.md

```diff
@@ -0,0 +1,82 @@
+# Block-Level Access Lists Implementation In Reth
+
+The project will add [**Eip-7928**](https://eips.ethereum.org/EIPS/eip-7928) aka `Block-Level Access Lists` support in Reth.
+
+## Motivation
+
+EIP-7928 introduces **Block-Level Access Lists** (BALs), which record all accounts and storage locations accessed during block execution, along with their corresponding post-execution values. 
+
+**Benefits Of Including Eip-7928 in Reth:**
+
+* BALs provide a complete view of which storage slots are accessed and by which transactions. This enables safe and efficient parallel execution and disk operations—more reliable than optimistic parallelization techniques.
+
+* Since BALs include full post-execution state diffs, the post-state can be reconstructed directly from the pre-state and BALs without re-executing transactions.
+
+* BALs contribute to a partially stateless Ethereum by preserving only the required subset of state data.
+
+* In zkEVM environments, BALs allow nodes to verify execution proofs and skip actual transaction execution while still maintaining full state integrity.
+
+* BALs allow account balance tracking without the need for extensive RPC calls.
+
+## Project Description
+
+This project will allow creation of Block-Level Access List that which will mark an integral step towards stateless ethereum.We will try to implement it as per the changes listed in eip-7928 and its execution specs.
+
+## Specification
+The specification will include adding `bal_hash` to the block header and `BlockAccessList` to the block body. This involves creation of `StorageChange`,`BalanceChange`,`NonceChange`,`CodeChange`,`SlotChanges`,`SlotRead`,`AccountChanges` for BAL. Modification to `state_transition` function to validate the block-level access lists and adding `build_block_access_list`.
```

Comment:
Specs are pretty short and vague here, might be worth expanding a bit with some context. 

It should give some insight into the design of your implementation based on reth architecture, maybe a diagram or something. Will you modify revm directly to get the data? Or will you use any revm `Inspector` to hook into the execution process from the outside? How is the access list stored, how is it serialized?